ingestors/support/encoding.py

```diff
@@ -18,37 +15,32 @@ class EncodingSupport(object):
     def decode_string(self, text, encoding=DEFAULT_ENCODING):
         if not isinstance(text, bytes):
             return stringify(text)
-        encoding = normalize_encoding(encoding)
+        encoding = tidy_encoding(encoding)
         try:
             return text.decode(encoding, "strict")
         except Exception:
             try:
-                detected = guess_encoding(text)
+                detected = predict_encoding(text)
                 return text.decode(detected, "strict")
             except Exception:
                 return text.decode(encoding, "replace")
 
     def detect_stream_encoding(self, fh, default=DEFAULT_ENCODING):
-        return guess_file_encoding(fh, default=default)
+        return predict_file_encoding(fh, default=default)
 
     def detect_list_encoding(self, items, default=DEFAULT_ENCODING):
-        detector = chardet.UniversalDetector()
         for text in items:
             if not isinstance(text, bytes):
                 continue
-            detector.feed(text)
-            if detector.done:
-                break
 
-        detector.close()
-        return normalize_result(detector.result, default)
+            return predict_encoding(text=text, default=DEFAULT_ENCODING)
```

Comment:
This looks odd to me. Earlier we fed the detector multiple text snippets till it made a definitive decision. The changed code makes a decision on the first text snippet only.